### ISSUE TITLE
Expanding the wiki (Ideas)

### ISSUE THREAD

**Issue by VehementHam** [Ultramarine-Linux/wiki-new#35]
# Expanding the wiki
I have put together a list of ideas in case the commity of Ultrumarine Linux wishes to expand the wiki.

First of all, I'd like to say, that I think the idea of a centralized wiki where the user can go and learn about their system, is a genius idea, because all of the information is optimized for Ultramarine Linux specifically. That is what this community has accomplished.

Because this wiki is very well-made, and thought out, I think that the users would benifit from more of it. If the community was to expand the wiki, here are some pages ideas that I would think could be a prority:
## git
`git` is one of the most difficult commands in Linux. It would be extremely benifitial if the user could conveniently, and properly learn how to use it in this wiki. This page would provide instructions on how Git works, and how to use it from the command line. It would also go in depth on how to properly authenticate with ssh tokens and whatnot. It will go over common issues that users have. All while remaining breif and consise. If users wish to learn more about Git, the page will have links to the Pro Git book, and the Github docummentation.
## ssh
Another important command on Linux which users find scary. The page would include information on connecting to other systems via `ssh`, and mounting a system with `sshfs`. It would also include configuration instructions.
## Installing applications
Detailed instructions on the different ways to install applications. This includes flatpaks, snaps, appimages, nix home manager, pkgsrc, distrobox containers, and an in depth look at Fedoras package management system; copr, rpms, dnf.
## Disk management
An explanation on disk management. Details on how to partition, mount, create and change file sytsems. Instructions for gparted, and command line applications.
## gpg
I don't know much about this one, but I know that it is a scary one. I think users would benefit from understanding GPG encryption.
## grep awk sed etc.
More advanced but useful shell commands. This one would be very easy to write like `shell.mdx`.
## shell scripting
A very breif guide on the basics of ZSH and SH scripting. Will save users hours if they learn it.
## Package specific doccumentation stubs
The Arch and Gentoo wiki have docummentation/guides on specific packages. I would be very happy to write little guides on individual Linux commands.
### dd
This can be added to the section that explains how to install Ultramarine Linux. I may also be added to the Disk Management section instead. Or the package specific docummentation. When I was a new user, I found this command very scary. 
### Vis
Vim/Vis is a very scary thing to new users. I would write a page on this. It would go under package specific doccummentation.

### GNU Pass
Another package specific. This one in particular is very scary.
## Recommended Reading
I was thinking that on each page, the community could recommend a book for the user to read to gain a more in depth knoledge of the topic.
## Security
A page that explains cyber security basics. A topic that is very much relivent to using and learning Linux.

##List of applications
I was also thinking that we could add a list of applications for different purposes. Users are always looking up "What is the best image veiwer for Linux," or something. This list would give them options. I have created a basic example of such a list. Each application would have a link and a discription as well.
### Browsers
* Floorp
* Qutebrowser
* Mothra
* Lynx
### Adblocking
Ublock Origin
[/etc/hosts](https://surf.suckless.org/files/adblock-hosts/)
### Spreadsheet
Gnumeric
LibreOffice
Only Office
[Awk](https://pspodcasting.net/dan/blog/2019/plan9_desktop.html#office_db_awk)
### Text editors
VSCodium
Vis
Nvchad
### Games
0A.D
Warmux
Supertux Kart
Supertux
Supertux2
Minetest
Freeciv
Veloren
### Application launchers
Dmenu
Fuzzel
Rofi
### Terminal Emulators
ST
Foot
Kitty
Budgie's default terminal
### Image viewers
Gwenview
Imv
Feh
Ristretto
### Volume control
pactl
pavucontrol
### Caulculator
kcalc
lua
gforth
### Unit conversion
Convertall
### Media Player
mpv
VLC
Haruna
yt-dlp
### File sharing clients
nicotine+
qbittorrent
### Password managers
KeePassXC
GNU Pass
### File Managers
Dolphin
Krusader
Pcmanfm
Rox filer
SFM
Ranger
### Compilers
GCC
TCC
Clang/LLVM
Cproc/QBE
# Conclusion
Anyways, those are the ideas that I came up with last night while I was heading to bed. I don't know the philisophy of the wiki, but if the community wants to expand it, here are ideas.

**Comment by VehementHam** [comment]
By the way, whatever applicatons are included in Ultramarine inux by default would have a special mark in the list of applicaton page

**Comment by nothingneko** [comment]
Security page is actually in progress right now. Applications page seems incredibly excessive. The book thing is a bit iffy, we kinda want the wiki to be a fast reference grab and go type deal. some of the more basic commands may be helpful, but we're moving into more advanced territory with the scripting instructions and expanding the software installation page.

**Comment by VehementHam** [comment]
Let me clarify a few things here. 

> Application page seems incredably excessive.

I just watched a video, made by a friend named Waffles (he is the best Linux Youtuber). He is an experinced Linux user, he knows more than me in a lot of areas that have to do with the Linux desktop, and whatnot. One of which is knowing the different applications out there.

In this video he states that one of the most difficult things for new Linux users is finding these applications, and that new Linux users will always have to their own research. At that moment, I saw the potential benifit of a page like this. It would be there, along with all of the other recources that are supposed to help new users get familiar with their systmes. 

Waffles said that he had just now found the right image viewer and music player after a long time of searchig. I think that UM users could benifit from seeing all their options (that they didn't even know to look for) listed out plainly on that page. The description/link is there too.

> The book thing is a bit iffy

There are links to "The Linux Shell in 100 Seconds," and whatnot on the wiki page. Why not have a reading link too? Maybe that 100 second video is not enough.

> we kinda want the wiki to be a fast reference grab and go type deal.

That does not mean that the wiki needs to be small. If the community has a desire to contribute to more wiki pages, why turn it down? Just put the extra stuff out of the users way until they finish with the grab and go page It's all about organization. The current "grab and go" wiki can be it's own section. Then we make another section for slightly more in depth concetps.

Nothing about adding more to the wiki nessessarily takes away the "fast" nature. I think that the wiki is really good, and it would be blowing an opportunity not to make something better and bigger of it. I agree that the fast and easy introduction into Linux is great, but where do users go after their introduction?

> into more advanced territory with the scripting instructions

I said that it would be "easy" in my issue for a reason. This is not a C course. Shell scripting is by nature, really simple, and requires a small introduction, no more. The page would talk about `while` loops, `if` statemetns, oporators, variables, and command arguments. That would be enough to get the user scripting, and saving time. There would be a book link for users who want to go further. There should always be somewhere for the user to go aftere reading, it doesn't have to be another wiki page.

> when you start mixing in a third package manager (like nix) things get really messy really quickly

Same thing as making second `shell.mdx` page. for `grep`, `awk`, and `sed`. Other package managers/installation methods are simply a continuation of the page for users who want to learn more. If this distro is intending on being the ultimate new user distro, it should have an extensive wiki. Also compiling stuff from source is extremely difficult for new users, and they would benifit from a guide.

Application specific pages could be the links on the list of application pages. No need to clutter the contents with them.

My conclusion is to restate this point: _If the community has a desire to contribute to more wiki pages, why turn it down?_

Of course if the Ultra Marine team has no intention of expanding/putting more effort into the wiki, then this can all be forgotten. This issue is mearly an idea of how to add more.

**Comment by nothingneko** [comment]
> difficult things for new Linux users is finding these applications, and that new Linux users will always have to their own research.
This should be dealt with in the app store, not on the wiki. We need more curation.

We can talk more about how to do something like this in the wiki, but I feel that it's not the proper place for this

> There are links to "The Linux Shell in 100 Seconds," and whatnot on the wiki page. Why not have a reading link too? Maybe that 100 second video is not enough.

thanks for clarifying, much more reasonable. Just concerned about users needing to purchase things

> That does not mean that the wiki needs to be small.
We do still need to discuss how we want to handle pages beyond the currently very guidey format

> said that it would be "easy" in my issue for a reason.
Alright, this seems reasonable to me

> Other package managers/installation methods are simply a continuation of the page for users who want to learn more

perhaps we should have an "advanced" or "further learning" link on these pages that brings you to a page not shown in the sidebar with more in-depth information.

**Comment by VehementHam** [comment]
> thanks for clarifying, much more reasonable. Just concerned about users needing to purchase things

IG, we would only link open source ebooks, like the Pro Git book. There are enough out there.

**Comment by nothingneko** [comment]
> > thanks for clarifying, much more reasonable. Just concerned about users needing to purchase things
> 
> IG, we would only link open source ebooks, like the Pro Git book. There are enough out there.

seems good to me, go right ahead with that when you edit

what do you think of

> perhaps we should have an "advanced" or "further learning" link on these pages that brings you to a page not shown in the sidebar with more in-depth information.

**Comment by VehementHam** [comment]
> what do you think of

> > perhaps we should have an "advanced" or "further learning" link on these pages that brings you to a page not shown in the sidebar with more in-depth information.

That is sort of what I had in mind. I think that's similar to what other wikis are doing.

**Comment by nothingneko** [comment]
alright, you're more than welcome to implement that in your writing. i think this mostly fits in with the linux concepts section (which you are more than welcome to expand) the hardware and installation sections currently benefit from behaving more like a guide so let's leave those be for now

**Comment by lleyton** [comment]
I'm generally in favor of more pages in the wiki, if the community wants it :p

if the amount of content gets unmanageable, worst case, we just split it into a "in-depth" page and hide it from the sidebar by default

TL;DR: go ahead, I'm excited to see where this goes
